presto-native-execution/presto_cpp/main/PrestoServer.h

```diff
@@ -115,16 +116,14 @@ class PrestoServer {
   void enableAnnouncer(bool enable);
 
  protected:
+  virtual void createPeriodicMemoryChecker();
```

Comment:
Why not encapsulate the init and stop in
void startPeriodicMemoryChecker()
void stopPeriodicMemoryChecker()

Reply:
Maybe we can make this an empty virtual? The implementing PrestoServer will fill it up with the needed checker, like other virtual functions.

Reply:
> Why not encapsulate the init and stop in void startPeriodicMemoryChecker() void stopPeriodicMemoryChecker()

The idea is to have only one overridable method: create().
Then only the base class need to call start() and stop() for the checker if it exists.
No need to have two methods overridable.

Any class inheriting from Server simply needs to override one method of creating the checker.

Reply:
> Maybe we can make this an empty virtual? The implementing PrestoServer will fill it up with the needed checker, like other virtual functions.

I was planning that, but the problem is backed support for a define into the project and creation of LinuxMemoryChecker beind the scenes. Check the code.

So can't do what you propose.

Reply:
The macro based singleton checker pick is quite unintuitive. Since now we are storing the checker in server, can we remove the singleton and macro based checker pick. We might need to collaborate with oss community on this so this does not break their deployment.

Reply:
> The macro based singleton checker pick is quite unintuitive. Since now we are storing the checker in server, can we remove the singleton (which is removed from this PR) and macro based checker pick. We might need to collaborate with oss community on this so this does not break their deployment.

I don't like macro either.
But changing this is outside of the scope of this diff.
I made the change as less intrusive as possible.
We can follow up removing the define and maybe make it based on config or something else at any time later.